internal/jobcontainers/process.go

```diff
@@ -0,0 +1,179 @@
+package jobcontainers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"golang.org/x/sys/windows"
+)
+
+// JobProcess represents a process run in a job object.
+type JobProcess struct {
+	cmd            *exec.Cmd
+	procLock       sync.Mutex
+	stdioLock      sync.Mutex
+	stdin          io.WriteCloser
+	stdout         io.ReadCloser
+	stderr         io.ReadCloser
+	waitBlock      chan struct{}
+	closedWaitOnce sync.Once
+	waitError      error
+}
+
+var _ cow.Process = &JobProcess{}
+
+func newProcess(cmd *exec.Cmd) *JobProcess {
```

Comment:
Am I missing how the process here is hooked up to run in a job object? I don't see any other calls to `CreateJobObject` in this PR.

Reply:
It's called in `CreateContainer` thats not in this PR. During `CreateProcess` (method off of JobContainer) it's assigned to the job object

Reply:
Would it make more sense to split `jobcontainers` completely out of this PR? It seems this is implementing an incomplete set of that package. What's the rationale for what goes into which PR?

Reply:
Yea we could, I guess there's no difference in it being excluded from this. I mostly had it in so I could run tests against the jobProcess methods.